operator/pkg/apis/core/v1beta1/configconnectorcontext_types.go

```diff
@@ -75,6 +77,22 @@ func init() {
 	SchemeBuilder.Register(&ConfigConnectorContext{}, &ConfigConnectorContextList{})
 }
 
+var _ addonv1alpha1.CommonObject = &ConfigConnectorContext{}
```

Comment:
It's the common idiom to express that the struct implements the interface.  There's an argument about whether it should be done at all if e.g. there's an implicit cast to the interface like in `func New() Interface { return &Struct{} }`, but my take is that explicit is better.

Here it seems important, and we aren't catching it otherwise.  My view is that one bug is enough to settle the argument about whether it's a good idea in general, but I haven't won that argument :-)

Reply:
Thank you for sharing the context! I like the idea of making it explicit especially for readability purpose.